### DISCUSSION TITLE
Configure client for Azure and OpenAI in parallel

### DISCUSSION THREAD

**Discussion by remixtedi** [betalgo/openai#251] [Q&A] [answered]
If the client is configured to use Azure OpenAI service then Image generation doesn't work because it is not available on Azure.
I wonder if there is a way to have the client configured for both platforms? to use Azure for chat completions and OpenAI Dalle for images?

**Comment by magols** [comment]
The same goes for embeddings. You get different deployment IDs for the different models deployed in Azure but there is only one setting for that in the config. Also the API-versions are different.

So it will call GPT instead  of Ada of its configured to work with chat completions

**Comment by kayhantolga** [comment, marked as answer]
This is a case that I haven't considered before. To address it, I will add support for multi-instance initialization. As a workaround, you can create a second instance without using dependency injection.

**Comment by kayhantolga** [comment]
https://github.com/betalgo/openai/wiki/Working-with-Multiple-Instances wiki page is ready, I will be releasing 6.9.0 soon

This update will have only support for typed multiple instances but I am also planning dynamic multiple instances in the next versions.

**Reply by kayhantolga** [reply]
https://github.com/betalgo/openai/pull/258
